Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -49,52 +52,291 @@ public FunctionManager(IEnvironment environment, ITraceFactory traceFactory)
+            return await GetHostConfig();
+        }
+
+        public FunctionEnvelope CreateFunctionConfig(string configContent, string functionName)
+        {
+            var config = JObject.Parse(configContent);
+            return new FunctionEnvelope
+            {
+                Name = functionName,
+                ScriptRootPathHref = FilePathToVfsUri(GetFunctionPath(functionName), isDirectory: true),
+                ScriptHref = FilePathToVfsUri(GetFunctionScriptPath(functionName, config)),
+                ConfigHref = FilePathToVfsUri(Path.Combine(GetFunctionPath(functionName), Constants.FunctionsConfigFile)),
+                TestDataHref = FilePathToVfsUri(GetFunctionSampleDataFile(functionName)),
+                SecretsFileHref = FilePathToVfsUri(GetFunctionSecretsFile(functionName)),
+                Href = GetFunctionHref(functionName),
+                Config = config
+            };
+        }
+
+        public IEnumerable<FunctionTemplate> GetTemplates()
+        {
+            return new[]
+            {
+                new FunctionTemplate { Id = "HttpTrigger", Language = "JavaScript", Trigger = "Http" },
+                new FunctionTemplate { Id = "HttpTrigger-Batch", Language = "Batch", Trigger = "Http" },
+                new FunctionTemplate { Id = "BlobTrigger", Language = "JavaScript", Trigger = "Blob" },
+                new FunctionTemplate { Id = "QueueTrigger-Bash", Language = "Bash", Trigger = "Queue" },
+                new FunctionTemplate { Id = "QueueTrigger-Batch", Language = "Batch", Trigger = "Queue" },
+                new FunctionTemplate { Id = "QueueTrigger-FSharp", Language = "F#", Trigger = "Queue" },
+                new FunctionTemplate { Id = "QueueTrigger-Php", Language = "Php", Trigger = "Queue" },
+                new FunctionTemplate { Id = "QueueTrigger-Powershell", Language = "PowerShell", Trigger = "Queue" },
+                new FunctionTemplate { Id = "QueueTrigger-Python", Language = "Python", Trigger = "Queue" },
+                new FunctionTemplate { Id = "QueueTrigger", Language = "JavaScript", Trigger = "Queue" },
+                new FunctionTemplate { Id = "ResizeImage", Language = "exe", Trigger = "Queue" },
+                new FunctionTemplate { Id = "ServiceBusQueueTrigger", Language = "JavaScript", Trigger = "ServiceBus" },
+                new FunctionTemplate { Id = "TimerTrigger", Language = "JavaScript", Trigger = "Timer" },
+                new FunctionTemplate { Id = "WebHook-Generic", Language = "JavaScript", Trigger = "WebHook-Generic" },
+                new FunctionTemplate { Id = "WebHook-GitHub", Language = "JavaScript", Trigger = "WebHook-GitHub" }
+            };
+        }
+
+        public void DeleteFunction(string name)
+        {
+            FileSystemHelpers.DeleteDirectorySafe(GetFunctionPath(name), ignoreErrors: false);
+            FileSystemHelpers.DeleteFileSafe(GetFunctionSampleDataFile(name));
```

Comment:
@davidebbo What do you think the right behavior here should be? Should we delete the log directory for a function when that function is deleted? I think it makes sense.